text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,457 @@
+                1. Adds a step but doesn't integrate directly with CDK’s project generation lifecycle.
+    3. Yeoman Generator
+        1. Pros:
+            1. Highly customizable scaffolding logic
+                1. Enables conditional file inclusion, dynamic file naming, and complex transformations based on user input beyond static substitution like %name.PascalCased%.
+            2. Interactive CLI experience for users
+                1. Users can be prompted during `cdk init` to select options, allowing a single generator to produce many variations of a template.
+            3. Facilitates reusable and adaptable enterprise templates
+                1. Organizations can standardize internal CDK setups with customizable templates that adjust based on use case
+            4. Supports post-processing automation
+                1. After scaffolding, Yeoman can automatically run `npm install`, initialize Git, and add license headers
+        2. Cons:
+            1. Increased complexity and learning curve
+                1. CDK users would need to install and learn Yeoman
+                2. Template authors must write JS-based generators instead of simple file templates
+            2. External dependency
+                1. Relies on the Yeoman runtime and ecosystem so CDK loses control over some UX
+                2. Added maintenance risk from external changes
+            3. Not natively integrated into CDK CLI
+                1. `cdk init` logic needs to be updated to recognize and delegate to Yeoman generators
+                2. Makes CDK’s bootstrapping process more complex
+            4. Overkill for simple use cases
+                1. Most CDK users just want to get started quickly
+                2. %name.PascalCased% and static templates are sufficient
+    4. Projen
+        1. Pros: 
+            1. Controlled and maintained by AWS
+                1. Easier to add CDK specific behaviors
+            2. Automates common setup tasks (like linting, testing, and publishing configs)
+                1. Template authors don’t have to maintain those manually across many templates.
+            3. One Projen-based template can generate consistent setups across different language targets.
+        2. Cons: 
+            1. Steep learning curve
+                1. Template authors must understand how Projen works, including its configuration language
+            2. Templates are code, not files
+                1. Instead of writing/copying static template files, template authors must write Projen config code that generates those files.
+            3. Generated files are not meant to be edited
+                1. Generated files are overwrited on every synth, so template users should only modify the config and not the output file
+            4. Harder to preview template
+                1. Since the actual files aren’t visible until you run `projen synth`, don’t initially know what the final output will look like.
```

Comment:
Yes, but projen based projects run `projen synth` during build, and the generated files are checked into the repo.